sound/soc/soc-pcm.c

```diff
@@ -1137,7 +1137,7 @@ static int dpcm_be_connect(struct snd_soc_pcm_runtime *fe,
 			return 0;
 	}
 
-	dpcm = kzalloc(sizeof(struct snd_soc_dpcm), GFP_KERNEL);
+	dpcm = kzalloc(sizeof(struct snd_soc_dpcm), GFP_ATOMIC);
```

Comment:
reworded as 
"
We allocate a structure in dpcm_be_connect(), which may be called in
    atomic context. Using GFP_KERNEL is not quite right, we have to use
    GFP_ATOMIC to prevent the allocator from sleeping.
"